src/core/components/InstallButton/index.js

```diff
@@ -194,195 +185,148 @@ export class InstallButtonBase extends React.Component {
     });
   }
 
+  showLoadingAnimation() {
+    const { addon, status } = this.props;
+
+    if (addon.type === ADDON_TYPE_STATIC_THEME && status === INSTALLED) {
+      // We have to enable a static theme after having installed it, so we keep
+      // the animation visible to avoid flickering.
+      return true;
+    }
+
+    return [DOWNLOADING, ENABLING, INSTALLING, UNINSTALLING].includes(status);
+  }
+
+  getButtonText() {
+    const { addon, i18n, status } = this.props;
+
+    switch (status) {
+      case DISABLED:
+        return i18n.gettext('Enable');
+      case ENABLED:
+      case INSTALLED:
+        return i18n.gettext('Remove');
+      case ENABLING:
+        return i18n.gettext('Enabling');
+      case DOWNLOADING:
+        return i18n.gettext('Downloading');
+      case INSTALLING:
+        return i18n.gettext('Installing');
+      case UNINSTALLING:
+        return i18n.gettext('Uninstalling');
+      default:
+        return isTheme(addon.type)
+          ? i18n.gettext('Install Theme')
+          : i18n.gettext('Add to Firefox');
+    }
+  }
+
+  getIconName() {
+    const { status } = this.props;
+
+    switch (status) {
+      case DISABLED:
+        return 'plus-dark';
+      case ENABLED:
+      case INSTALLED:
+        return 'delete';
+      default:
+        return 'plus';
+    }
+  }
+
   render() {
     const {
+      _config,
+      _log,
       addon,
-      clientApp,
       className,
       defaultInstallSource,
-      getClientCompatibility,
-      hasAddonManager,
-      i18n,
+      disabled,
       location,
+      status,
       userAgentInfo,
-      _config,
-      _log,
-      _window,
     } = this.props;
 
     if (addon.type === ADDON_TYPE_OPENSEARCH && _config.get('server')) {
       _log.info('Not rendering opensearch install button on the server');
       return null;
     }
 
-    // OpenSearch plugins display their own prompt so using the "Add to
-    // Firefox" button regardless on mozAddonManager support is a better UX.
-    const useButton =
-      (hasAddonManager !== undefined && !hasAddonManager) ||
-      addon.type === ADDON_TYPE_OPENSEARCH ||
-      this.props.useButton;
-    let button;
-
-    const { compatible } = getClientCompatibility({
```

Comment:
Is this not needed anymore? I don't see it. Previously, I think this was used to gray out the buttons in certain scenarios.

Reply:
The parent component does that in both cases already. There is an issue about too many compatibility checks.